numpy/core/tests/test_api.py

```diff
@@ -38,58 +38,39 @@ def test_array_array():
         np.array(np.array(1.0, dtype=np.float64), dtype=np.float64)
         assert_equal(old_refcount, sys.getrefcount(np.float64))
 
+    # test string
+    B2 = np.dtype((bytes, 2))
+    B3 = np.dtype((bytes, 3))
+    B5 = np.dtype((bytes, 5))
+    assert_equal(np.array(b"1.0", dtype=np.float64),
+                 np.ones((), dtype=np.float64))
+    assert_equal(np.array(b"1.0").dtype, B3)
+    assert_equal(np.array(b"1.0", dtype=bytes).dtype, B3)
+    assert_equal(np.array(b"1.0", dtype=B2), np.array(b"1."))
+    assert_equal(np.array(b"1", dtype=B5), np.ones((), dtype=B5))
+
     # test string
     S2 = np.dtype((str, 2))
     S3 = np.dtype((str, 3))
     S5 = np.dtype((str, 5))
-    assert_equal(np.array("1.0", dtype=np.float64),
+    assert_equal(np.array(str("1.0"), dtype=np.float64),
                  np.ones((), dtype=np.float64))
-    assert_equal(np.array("1.0").dtype, S3)
-    assert_equal(np.array("1.0", dtype=str).dtype, S3)
-    assert_equal(np.array("1.0", dtype=S2), np.array("1."))
-    assert_equal(np.array("1", dtype=S5), np.ones((), dtype=S5))
-
-    # test unicode
-    _unicode = globals().get("unicode")
-    if _unicode:
-        U2 = np.dtype((_unicode, 2))
-        U3 = np.dtype((_unicode, 3))
-        U5 = np.dtype((_unicode, 5))
-        assert_equal(np.array(_unicode("1.0"), dtype=np.float64),
-                     np.ones((), dtype=np.float64))
-        assert_equal(np.array(_unicode("1.0")).dtype, U3)
-        assert_equal(np.array(_unicode("1.0"), dtype=_unicode).dtype, U3)
-        assert_equal(np.array(_unicode("1.0"), dtype=U2),
-                     np.array(_unicode("1.")))
-        assert_equal(np.array(_unicode("1"), dtype=U5),
-                     np.ones((), dtype=U5))
+    assert_equal(np.array(str("1.0")).dtype, S3)
```

Comment:
```suggestion
    assert_equal(np.array("1.0").dtype, S3)
```
Etc, same for all the others. Equivalent and shorter.

Reply:
Done